File: google-code-jam/2013/round1a/A.java
import java.util.*;
import java.io.*;
public class A {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int tc = 1; tc <= T; tc++) {
            double r, t;
            r = sc.nextDouble();
            t = sc.nextDouble();
            long numEst = (long) ((-2 * r + 1 + Math.sqrt((2 * r - 1) * (2 * r - 1) + 8 * t)) / 4.0);
            long ans = -1;
            for (long num = numEst + 2; num >= 1; num--) {
                long tUsed = (long) (2 * num * num + (2 * r - 1) * num);
                if (tUsed <= t) {

                    break;
                }
            }
            System.out.printf("Case #%d: %d\n", tc, ans);
        }
    }
}

File: google-code-jam/2013/round1a/A.py
import math
T = int(raw_input())
for tc in range(1, T + 1):
    r, t = map(float, raw_input().split())
    numEst = int((-2 * r + 1 + math.sqrt((2 * r - 1) * (2 * r - 1) + 8 * t)) / 4.0)
    num = numEst + 1000
    while num > 0:
        if (2 * num * num + (2 * r - 1) * num) <= t:
            print 'Case #%d: %d' % (tc, num)
            break
        num -= 1
